### ISSUE TITLE
Start building an ML model

### ISSUE THREAD

**Issue by iwandejong** [COS301-SE-2024/Crop-Prediction-System#61]
Preprocessing
- [x] NDVI (image pentadal)
- [x] PET (image pentadal)
- [x] CHIRPS (image pentadal)
- [x] Cloud Cover (monthly)
- [x] Ground Frost Frequency (monthly)
- [x] Precipitation (monthly)
- [x] Temp Min (monthly)
- [x] Temp Max (monthly)
- [x] Temp Mean (monthly)
- [x] Vapour (monthly)
- [x] Raining Days (monthly)
- [x] Growing Season (Seasonal)
- [x] Crop Stage Index (generated based on department research)
- [x] Historic Crop Yields (various crops)

Combining everything:
- [x] Make a table in Supabase to combine all the statistics into one dataset (means and stdev for each data point)
- [x] Interpolate all the data to be pentadal
- [x] Correlation Tests
- [x] Statistical Feature Selection (R/Python)
- [ ] Make a document to describe the model so it can easily be worked on
- [x] Start building the model (regression, forest, gradient boosting, neural network - in research)

Progress: https://lucid.app/lucidchart/9f71ed4a-68f7-4a86-8a67-66fbcb5485c8/edit?viewport_loc=-770%2C-21%2C2820%2C1362%2C0_0&invitationId=inv_9de32da7-f419-411b-a439-5f48bd857737

**Comment by iwandejong** [comment]
- [ ] Data integration (storing of data)
- [ ] Data preprocessing (Cleaning, aligning and restoring)
- [ ] Exploratory Analysis (R)
- [ ] Statistical Feature Selection (R/Python)
- [ ] Start building the model (regression, forest, gradient boosting, neural network - in research)

**Comment by iwandejong** [comment]
All preprocessing facilities are available. Problem is time it takes to analyse all of them and store it in Supabase.

- PET: 40 minutes
- CHIRPS: 40 minutes
- NDVI: 14 hours

**Comment by iwandejong** [comment]
Built a basic random forest model on wheat. Seems to work pretty well, but will fine-tune as time progresses to ensure that there's no bias or other factors affecting the model
